infra/modules/bedrock-data-automation/variables.tf

```diff
@@ -0,0 +1,137 @@
+  type        = string
+  default     = null
+}
+
+variable "bda_kms_encryption_context" {
+  description = "The KMS encryption context for the Bedrock data automation project."
+  type        = map(string)
+  default     = null
+}
+
+variable "bda_kms_key_id" {
+  description = "The KMS key ID for the Bedrock data automation project."
+  type        = string
+  default     = null
+}
+
+variable "bda_custom_output_config" {
+  description = "A list of the BDA custom output configuartion blueprint(s)."
+  type = list(object({
+    blueprint_arn     = optional(string)
+    blueprint_stage   = optional(string)
+    blueprint_version = optional(string)
+  }))
+  default = null
+}
+
+variable "bda_standard_output_configuration" {
+  description = "Standard output is pre-defined extraction managed by Bedrock. It can extract information from documents, images, videos, and audio."
+  type = object({
+    audio = optional(object({
+      extraction = optional(object({
+        category = optional(object({
+          state = optional(string)
+          types = optional(list(string))
+        }))
+      }))
+      generative_field = optional(object({
+        state = optional(string)
+        types = optional(list(string))
+      }))
+    }))
+    document = optional(object({
+      extraction = optional(object({
+        bounding_box = optional(object({
+          state = optional(string)
+        }))
+        granularity = optional(object({
+          types = optional(list(string))
+        }))
+      }))
+      generative_field = optional(object({
+        state = optional(string)
+      }))
+      output_format = optional(object({
+        additional_file_format = optional(object({
+          state = optional(string)
+        }))
+        text_format = optional(object({
+          types = optional(list(string))
+        }))
+      }))
+    }))
+    image = optional(object({
+      extraction = optional(object({
+        category = optional(object({
+          state = optional(string)
+          types = optional(list(string))
+        }))
+        bounding_box = optional(object({
+          state = optional(string)
+        }))
+      }))
+      generative_field = optional(object({
+        state = optional(string)
+        types = optional(list(string))
+      }))
+    }))
+    video = optional(object({
+      extraction = optional(object({
+        category = optional(object({
+          state = optional(string)
+          types = optional(list(string))
+        }))
+        bounding_box = optional(object({
+          state = optional(string)
+        }))
+      }))
+      generative_field = optional(object({
+        state = optional(string)
+        types = optional(list(string))
+      }))
+    }))
+  })
+  default = null
+}
+
+variable "bda_override_config_state" {
+  description = "Configuration state for the BDA override."
+  type        = string
+  default     = null
+}
+
+variable "bda_tags" {
```

Comment:
The modules don't typically take arbitrary tags to apply, but get them through the `default_tags` block on the `aws` provider. I'm assuming the `awscc` provider doesn't support that?

If not, then probably reasonable to take a `tags` variable, and `/infra/{{app_name}}/service/bda.tf` should pass in `local.tags` as it's value.